sdk/translation/azure-ai-translation-document/src/samples/java/com/azure/ai/translation/document/GetDocumentStatus.java

```diff
@@ -0,0 +1,93 @@
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.util.BinaryData;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Sample for getting documents status
+ */
+public class GetDocumentStatus {
+    public static void main(final String[] args) {
+        String endpoint = System.getenv("DOCUMENT_TRANSLATION_ENDPOINT");
+        String apiKey = System.getenv("DOCUMENT_TRANSLATION_API_KEY");
+        AzureKeyCredential credential = new AzureKeyCredential(apiKey);
+        
+        DocumentTranslationClient documentTranslationClient = new DocumentTranslationClientBuilder()
+                .endpoint(endpoint)
+                .credential(credential)
+                .buildClient();
+
+        // BEGIN:GetDocumentStatus
+        SyncPoller<TranslationStatus, Void> response
+            = documentTranslationClient
+                .beginStartTranslation(
+                    new StartTranslationDetails(Arrays.asList(new BatchRequest(
+                        new SourceInput("https://myblob.blob.core.windows.net/sourceContainer")
+                            .setFilter(new DocumentFilter().setPrefix("pre").setSuffix(".txt"))
+                            .setLanguage("en")
+                            .setStorageSource(StorageSource.AZURE_BLOB),
+                        Arrays
+                            .asList(
+                                new TargetInput("https://myblob.blob.core.windows.net/destinationContainer1", "fr")
+                                    .setCategory("general")
+                                    .setGlossaries(Arrays.asList(new Glossary(
+                                        "https://myblob.blob.core.windows.net/myglossary/en_fr_glossary.xlf", "XLIFF")
+                                        .setStorageSource(StorageSource.AZURE_BLOB)))
+                                    .setStorageSource(StorageSource.AZURE_BLOB),
+                                new TargetInput("https://myblob.blob.core.windows.net/destinationContainer2", "es")
+                                    .setCategory("general")
+                                    .setStorageSource(StorageSource.AZURE_BLOB)))
+                        .setStorageType(StorageInputType.FOLDER))));
+        
+        String translationId = response.poll().getValue().getId(); 
+        
+        // Add Status filter
+        List<String> succeededStatusList = Arrays.asList(Status.SUCCEEDED.toString());
+        RequestOptions requestOptions = new RequestOptions();
+        requestOptions.addQueryParam("statuses",
+                succeededStatusList.stream()
+                    .map(paramItemValue -> Objects.toString(paramItemValue, ""))
+                    .collect(Collectors.joining(",")),
+                false); 
+        
+        try {
+            PagedIterable<BinaryData> documentStatusResponse = documentTranslationClient.getDocumentsStatus(translationId, requestOptions);
```

Comment:
We should rather use the `PagedIterable<DocumentStatus> getDocumentsStatus` APIs that give the model directly rather than using the `BinaryData` APIs and using mapper then to convert them. In tests and samples.
Especially in samples as they are easier to understand.

Reply:
Updated in Test code and samples